lxd/networks_config.go

```diff
@@ -0,0 +1,268 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/lxc/lxd/shared"
+)
+
+func networkValidPort(value string) error {
+	if value == "" {
+		return nil
+	}
+
+	valueInt, err := strconv.ParseInt(value, 10, 64)
+	if err != nil {
+		return fmt.Errorf("Invalid value for an integer: %s", value)
+	}
+
+	if valueInt < 1 || valueInt > 65536 {
+		return fmt.Errorf("Invalid port number: %s", value)
+	}
+
+	return nil
+}
+
+func networkValidAddress(value string) error {
+	err := networkValidAddressV4(value)
+	if err == nil {
+		return nil
+	}
+
+	err = networkValidAddressV6(value)
+	if err == nil {
+		return nil
+	}
+
+	return fmt.Errorf("Not a valid address: %s", value)
+}
+
+func networkValidAddressV6(value string) error {
+	if value == "" {
+		return nil
+	}
+
+	ip, net, err := net.ParseCIDR(value)
+	if err != nil {
+		return err
+	}
+
+	if ip.To4() != nil {
+		return fmt.Errorf("Not an IPv6 address: %s", value)
+	}
+
+	if ip.String() == net.IP.String() {
+		return fmt.Errorf("Not a usable IPv6 address: %s", value)
+	}
+
+	return nil
+}
+
+func networkValidNetworkV6(value string) error {
+	if value == "" {
+		return nil
+	}
+
+	ip, net, err := net.ParseCIDR(value)
+	if err != nil {
+		return err
+	}
+
+	if ip.To4() != nil {
+		return fmt.Errorf("Not an IPv6 network: %s", value)
+	}
+
+	if ip.String() != net.IP.String() {
```

Comment:
It does because I re-define "net" as a variable, will fix the variable name to make this more readable.